erpnext/stock/stock_ledger.py

```diff
@@ -1118,6 +1137,29 @@ def get_future_sle_with_negative_qty(args):
 		limit 1
 	""", args, as_dict=1)
 
+
+def get_future_sle_with_negative_batch_qty(args):
+	return frappe.db.sql("""
+		with batch_ledger as (
```

Comment:
I've thought a lot about this, there's no simple way to ensure performance + correctness both other than CTE + window function. LMK if anyone has a better suggestion.